tools/shared/src/main/scala/org/scalajs/core/tools/io/VirtualFiles.scala

```diff
@@ -131,6 +131,11 @@ trait VirtualSerializedScalaJSIRFile extends VirtualBinaryFile with VirtualScala
     try {
       ir.InfoSerializers.deserialize(stream)
     } catch {
+      case e: ir.IRVersionNotSupportedException =>
+        throw new IOException(
+          s"Failed to deserialize info of file compiled with Scala.js ${e.version}" +
+          s" (supported: ${e.supported.mkString(", ")}): $path", e)
```

Comment:
I think in this case it is clearly not an implementation detail, that we deserialize the underlying IR. Therefore, throwing an `IRVersionNotSupportedException` is probably OK.